<launch>
    <!-- Arguments -->
    <arg name="model" default="waffle_pi" />
    <arg name="bag_name" default="fixed_slam_easy" />
    <arg name="map_file" default="$(find turtlebot3_datasets)/data/map.yaml" />
    <arg name="open_rviz" default="true" />
    <arg name="initial_pose_x" default="0.000000" />
    <arg name="initial_pose_y" default="0.000000" />
    <arg name="initial_pose_a" default="0.000000" />

    <!-- Turtlebot3 -->
    <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
        <arg name="model" value="$(arg model)" />
    </include>

    <!-- Map server -->
    <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)" />

    <!-- TF -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="initial_tf" args="0.935 1.34 -0.023 0.001 -0.003 0.737 0.676 mocap map"/>

    <!-- Rosbag -->
    <param name="/use_sim_time" value="true" />
    <node pkg="rosbag" type="play" name="rosbag_player"
        args="--clock $(find turtlebot3_datasets)/data/$(arg bag_name).bag" />
    
    <!-- AMCL -->
    <node pkg="amcl" type="amcl" name="amcl">
        <param name="initial_pose_x" value="$(arg initial_pose_x)" />
        <param name="initial_pose_y" value="$(arg initial_pose_y)" />
        <param name="initial_pose_a" value="$(arg initial_pose_a)" />

        <param name="initial_cov_xx" value="2.0 * 2.0" />
        <param name="initial_cov_yy" value="2.0 * 2.0" />

        <param name="min_particles" value="500" />
        <param name="max_particles" value="5000" />

        <param name="odom_frame_id" value="odom" />
        <param name="base_frame_id" value="base_footprint" />
    </node>

    <!-- Estimated path -->
    <node name="amcl_path_publisher" pkg="turtlebot3_datasets" type="transformed_amcl.py"></node>
    <node name="ground_truth_publisher" pkg="turtlebot3_datasets" type="publish_gt.py"></node>
    <node name="amcl_error_saver_node" pkg="turtlebot3_datasets" type="publish_amcl_error.py"></node>
    <node name="data_saver_node" pkg="turtlebot3_datasets" type="amcl_npz_retriever.py"></node>

    <!-- rviz -->
    <node pkg="rviz" type="rviz" name="rviz" required="true"
        args="-d $(find turtlebot3_datasets)/data/amcl_localization.rviz" />

    
</launch>

